apps/backend/app/config.py

```
@@ -28,17 +31,39 @@ class Settings(BaseModel):
+    def validate_no_dummy_credentials(cls, v: str, info) -> str:
+        """Validate that dummy/test credentials aren't used in production."""
+        dummy_patterns = ["test-", "dummy-", "sk-test-", "placeholder"]
+        env = os.getenv("NODE_ENV", os.getenv("ENVIRONMENT", "development"))
+
+        if env.lower() == "production":
+            for pattern in dummy_patterns:
+                if pattern in v.lower():
+                    raise ValueError(
+                        f"Dummy credential detected in production environment for {info.field_name}. "
+                        f"Please set real credentials."
+                    )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)